### PR TITLE
feat: text field resizable

### PR DESCRIPTION
**Pull request by dtscalac**
# Description

Make VoicesTextField resizable.

## Screenshots

https://github.com/user-attachments/assets/faa54f75-a2fc-49e9-a3a2-7bfb17a7550c

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/widgets/text_field/voices_text_field.dart [hunk 1]
```diff
@@ -131,6 +133,7 @@ class _VoicesTextFieldState extends State<VoicesTextField> {
     final textTheme = theme.textTheme;
 
     final labelText = widget.decoration?.labelText ?? '';
+    final resizable = widget.maxLines == null && widget.minLines == null;
```
Comment (damian-molinski) [inline catalyst_voices/lib/widgets/text_field/voices_text_field.dart:136]: Does it make sense to have it guarded with parameter? I'm thinking about mobile where its not natural to have this resize. 

Maybe nullable `resizableEnabled` where by default it's checked against `CatalystPlatform.isDesktop`?

Comment (dtscalac) [reply catalyst_voices/lib/widgets/text_field/voices_text_field.dart:136]: Ok, I'll add explicit param for this. TextField doesn't allow to be resizable if minLines or maxLines is specified so we need to make sure they are null.

Comment (dtscalac) [reply catalyst_voices/lib/widgets/text_field/voices_text_field.dart]: I made this param true by default is we're on CatalystPlatform.isDesktop or CatalystPlatform.isWebDesktop

**Review by damian-molinski — APPROVED** [review]
LGTM!

**Pull request merged**